pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -818,9 +817,9 @@ def generate_samples(self):
             array[int]: array of samples in binary representation with shape
             ``(dev.shots, dev.num_wires)``
         """
-        return self.measurements.generate_samples(len(self.wires), self.shots).astype(
-            int, copy=False
-        )
+        shots = self.shots if isinstance(self.shots, int) else self.shots.total_shots
```

Comment:
```suggestion
        shots = self.shots if shots is None else shots
        shots = shots.total_shots if isinstance(shots, qml.measurements.Shots) else shots
```

Reply:
`generate_samples` in `lightning_gpu` does not take shots as input. I am a bit confused by this suggestion.

Reply:
nwm -- not required for lgpu